tests/src/providers/testqgspostgresconn.cpp

```diff
@@ -38,6 +40,35 @@ class TestQgsPostgresConn: public QObject
       QCOMPARE( QgsPostgresConn::quotedValue( "b \"c' \\x" ), QString( "E'b \"c'' \\\\x'" ) );
     }
 
+    void quotedValueDatetime()
+    {
+      QCOMPARE( QgsPostgresConn::quotedValue( QDateTime::fromString( "2020-05-07 17:56:00", "yyyy-MM-dd HH:mm:ss" ) ), QString( "'2020-05-07T17:56:00.000'" ) );
+
+      QgsFields fields;
+      QgsField f;
+      f.setName( "ts" );
+      f.setType( QVariant::DateTime );
+      f.setTypeName( "timestamp" );
+      fields.append( f );
+      QgsField f2;
+      f2.setName( "pk" );
+      f2.setType( QVariant::ULongLong );
+      f2.setTypeName( "serial8" );
+      fields.append( f2 );
+
+      QList<int> pkAttrs;
+      pkAttrs.append( 0 );
+      pkAttrs.append( 1 );
+
+      QgsPostgresSharedData *shared = new QgsPostgresSharedData;
+      QVariantList qvlist;
+      qvlist.append( QDateTime::fromString( "2020-05-07 17:56:00", "yyyy-MM-dd HH:mm:ss" ) );
+      qvlist.append( 123ULL );
+      shared->insertFid( 1LL, qvlist );
+
+      QCOMPARE( QgsPostgresUtils::whereClause( 1LL, fields, NULL, QgsPostgresPrimaryKeyType::PktFidMap, pkAttrs, std::shared_ptr<QgsPostgresSharedData>( shared ) ), QString( "\"ts\"='2020-05-07T17:56:00.000' AND \"pk\"=123" ) );
```

Comment:
Isn't `2020-05-07T17:56:00.000` incorrect here? In the PR description you state that you're trying to match Postgres' `yyyy-MM-dd HH:mm:ss` format.

Reply:
You are right! This is what I wrote at the beginning, I should have correct it. 

In the previous version of the code, the generated sql was: 
```... WHERE ("time"::text='2017-02-20T13:41:47.000' OR ...``` and the problem is the cast to ```::text```. 

In my various tests to check what kind of date can be handled by Postgresql, I made the mistake to believe only ```yyyy-MM-dd HH:mm:ss``` is supported but it can also handle ```yyyy-MM-ddTHH:mm:ss.SSS```. I was not able to make it read the same format but with timezone information like ```yyyy-MM-ddTHH:mm:ss.SSSZzz```. 

The simplest and most compatible format is ```Qt::ISODateWithMs``` ie. ```yyyy-MM-ddTHH:mm:ss.SSS```. I forced it to use this format because the default ```QDateTime.toString()``` differs from my pc (Qt5) and the one running the test on Github (?!?).